src/main/kotlin/open/tresorier/services/OperationService.kt

```diff
@@ -57,4 +60,58 @@ class OperationService(private val operationDao: IOperationDao, private val auth
         authorizationService.cancelIfUserIsUnauthorized(person, budget)
         return operationDao.findByBudget(budget)
     }
-}
+    fun openAndReadOfxFile(person: Person, account: Account, fileOfx: String) {
+        val formattedFile: String = fileOfx.replace("\n", "")
+        val doubleFormattedFile: String = formattedFile.replace("\r", "")
+        this.separateOperationsOfOfxFile(person, account, doubleFormattedFile)
+    }
+    fun separateOperationsOfOfxFile(person: Person, account: Account, ofxFile: String) : List<String> {
+        authorizationService.cancelIfUserIsUnauthorized(person, account)
+        var operationList = mutableListOf<String>()
+        var startTag = ofxFile.indexOf("<STMTTRN>")
+        var closeTag = ofxFile.indexOf("</STMTTRN>") + 10
+        var lastTag = ofxFile.lastIndexOf("</STMTTRN>") + 10
+        while (closeTag !== lastTag) {
+            operationList.add(ofxFile.substring(startTag, closeTag))
+            startTag = closeTag
+            closeTag = ofxFile.indexOf("</STMTTRN>", startTag) + 10
+        }
+        if (startTag !== -1) {
+            operationList.add(ofxFile.substring(startTag, closeTag))
+        }
+        this.passInEachOperation(person, account, operationList)
+        return operationList
```

Comment:
you return an operation list but it is unused, I think operation List may be confusing cause I thought the transformation to operation.class was done but it is not. Maybe ofxOperationList ?
+ this separate function also parse and add the list to the db => too much responsability.
Idealy you should only separate and return the list, the caller can then go on with the rest of the process

Reply:
this return is used in tests